app/Repositories/TripRepository.php

```diff
@@ -26,4 +26,29 @@ public function save(Trip $trip)
 
         return $trip;
     }
+
+    /**
+     * Update trip
+     *
+     * @param Trip $trip
+     * @param $attributes
+     * @return Trip
+     */
+    public function updateTrip(Trip $trip, $attributes)
```

Comment:
repository shouldn't fill your model. It should be able to save your model and find models in data set.
eg. `public function save(Trip $model): Trip`